File: app/src/main/java/com/bartovapps/gpstriprec/displayers/HmsDisplayer.java
package com.bartovapps.gpstriprec.displayers;

import android.widget.TextView;

public class HmsDisplayer implements TimeDisplayer {

	int seconds;
	int minutes;
	int hours;
	@Override
	public void displayTime(TextView view, long millis) {

		seconds = (int) (millis / 1000) % 60 ; 
		minutes = (int) ((millis / (1000*60)) % 60);
		hours   = (int) ((millis / (1000*60*60)) % 24);
		
		view.setText((hours>0 ? String.format("%d:", hours) : "") + ((this.minutes<10 && this.hours > 0)? "0" + String.format("%d:", minutes) :  String.format("%d:", minutes)) + (this.seconds<10 ? "0" + this.seconds: this.seconds));
	}

}

File: app/src/main/kotlin/presentation/displayers/HmsDisplayer.kt
package com.bartovapps.gpstriprec.presentation.displayers

import android.widget.TextView
import java.util.Locale
import javax.inject.Inject

class HmsDisplayer @Inject constructor(): TimeDisplayer {
    override fun displayTime(view: TextView, millis: Long) {
        val seconds = (millis / 1000).toInt() % 60
        val minutes = ((millis / (1000 * 60)) % 60).toInt()
        val hours = ((millis / (1000 * 60 * 60)) % 24).toInt()

        view.text = (if (hours > 0) String.format(
            Locale.getDefault(),
            "$hours:",
        ) else "") + (if ((minutes < 10 && hours > 0)) "0" + String.format(
            "%d:",
            minutes
        ) else String.format(Locale.getDefault(),
            "%d:",
            minutes
        )) + (if (seconds < 10) "0" + seconds else seconds)
    }
}